pywhy_graphs/algorithms/generic.py

```diff
@@ -565,6 +565,13 @@ def inducing_path(G, node_x: Node, node_y: Node, L: Set = None, S: Set = None):
     if node_x == node_y:
         raise ValueError("The source and destination nodes are the same.")
 
+    edges = G.edges()
+
+    for elem in edges.keys():
+        if elem not in {"directed", "bidirected"}:
+            if len(edges[elem]) != 0:
+                raise ValueError("Inducing Path is not defined for this graph.")
```

Comment:
I would instead do something like:

```Python
if any(edge_type not in {'directed', 'bidirected'} for edge_type in G.edge_types):
     raise...
```

See: https://www.pywhy.org/pywhy-graphs/dev/generated/pywhy_graphs.networkx.MixedEdgeGraph.html#pywhy_graphs.networkx.MixedEdgeGraph

Reply:
@adam2392 If I use this test all of the ADMG classes are raising errors as well.

Reply:
Oh okay I see. Can you add a comment marking this as something we should fix in the future `# XXX: fix this when graph classes are refactored to only check for directed/bidirected edge types`?